File: 3BIN/BINV3140-1 Programmation - question spéciales/BINV3140-A .NET/Séance 1-20200919/2. Exercices/BDActeurs/BDActeurs/src/be/ipl/film_database/domaine/Acteur.java
package be.ipl.film_database.domaine;

import java.io.Serializable;
import java.util.ArrayList;
import java.util.Calendar;
import java.util.Iterator;
import java.util.List;

public class Acteur extends Personne implements Serializable {

	/**
	 * 
	 */
	private static final long serialVersionUID = 1L;
	private final int tailleEnCm;
	private List<Film> films;

	public int getTailleEnCm() {
		return tailleEnCm;
	}

	
	
	public Acteur(String nom, String prenom, Calendar dateDeNaissance, int tailleEnCm) {
		super(nom, prenom, dateDeNaissance);
		this.tailleEnCm = tailleEnCm;
		films = new ArrayList<Film>();
	}



	@Override
	public String toString() {
		return "Acteur [nom = " + getNom() + " prénom = " + getPrenom() + " tailleEnCm=" + tailleEnCm + ", toString()="
				 + "]";
	}
	
	/**
	 * 
	 * @return La liste des films dans lequel l'acteur à joué.
	 */
	public Iterator<Film> films(){
		return films.iterator();
	}
	
	/**
	 * Ajoute le film à la liste des film dans lequel l'acteur a joué
	 * @param film
	 * @return false si le film est déjà présent ou null
	 */
	public boolean ajouterFilm(Film film){
		if ((film == null) || films.contains(film))
			return false;
		
		if (!film.contientActeur(this))
			film.ajouterActeur(this);
		
		films.add(film);
		
					
		
		return true;
	}
	
	/**
	 * 
	 * @param film
	 
	 */
	public boolean contientFilm(Film film){
		return films.contains(film);
	}
	
	/**
	 * @return le nom capitalisé
	 */
	@Override
	public String getNom(){
		return super.getNom().toUpperCase();
	}
}

File: 3BIN/BINV3140-1 Programmation - question spéciales/BINV3140-A .NET/Séance 1-20200919/3. Solution/BDActeursSolution/BDActeursSolution/ConsoleApplication1/Acteur.cs
using System;
using System.Collections.Generic;

namespace be.ipl.film_database.domaine
{
    [Serializable]
    class Acteur : Personne
    {
        private readonly int _tailleEnCm;
        private readonly List<Film> _films;

        public int GetTailleEnCm()
        {
            return _tailleEnCm;
        }



        public Acteur(String nom, String prenom, DateTime dateDeNaissance, int tailleEnCm) : base(nom, prenom, dateDeNaissance)
        {
            _tailleEnCm = tailleEnCm;
            _films = new List<Film>();
        }



        public override String ToString()
        {
            return "Acteur [nom = " + GetNom() + " prénom = " + GetPrenom() + " tailleEnCm=" + _tailleEnCm + ", toString()="
                     + "]";
        }

        ///
        ///
        /// <return> La liste des films dans lequel l'acteur à joué.</return>
        public IEnumerator<Film> FilmsIeEnumerator()
        {
            return _films.GetEnumerator();
        }

        /// <summary> Ajoute le film à la liste des film dans lequel l'acteur a joué </summary>
        /// <param> film </param>
        /// <return> false si le film est déjà présent ou null </return>
        public bool AjouterFilm(Film film)
        {
            if ((film == null) || _films.Contains(film))
                return false;

            if (!film.ContientActeur(this))
                film.AjouterActeur(this);

            _films.Add(film);



            return true;
        }

         /// <param> film</param>
         
        public bool ContientFilm(Film film)
        {
            return _films.Contains(film);
        }

        /// <return> le nom capitalisé </return>
        public override String GetNom()
        {
            return base.GetNom().ToUpper();
        }
    }
}
